timelock-agent/src/main/java/com/palantir/atlasdb/timelock/paxos/TimelockLeadershipMetrics.java

```diff
@@ -45,28 +48,30 @@ List<SafeArg<String>> namespaceAsLoggingArgs() {
                 SafeArg.of(AtlasDbMetricNames.TAG_CLIENT, proxyClient().value()));
     }
 
-    @Value.Derived
+    @Value.Lazy
     public PaxosLeadershipEventRecorder eventRecorder() {
         return PaxosLeadershipEventRecorder.create(
-                metrics().clientScopedMetrics().metricRegistryForClient(proxyClient()), // metrics for client etc.
+                taggedMetrics(), // metrics for client etc.
                 leaderUuid().toString(),
                 leadershipObserver(),
                 namespaceAsLoggingArgs());
     }
 
-    @Value.Derived
+    @Value.Lazy
     LeadershipObserver leadershipObserver() {
         return leadershipObserverFactory().create(proxyClient());
     }
 
-    public <T> T instrument(Class<T> clazz, T instance) {
-        return AtlasDbMetrics.instrumentWithTaggedMetrics(
-                metrics().clientScopedMetrics().metricRegistryForClient(proxyClient()),
-                clazz,
-                instance,
-                _context -> ImmutableMap.of(
-                        AtlasDbMetricNames.TAG_CURRENT_SUSPECTED_LEADER,
-                        String.valueOf(localPingableLeader().ping())));
+    @Value.Lazy
+    public Function<InvocationContext, Map<String, String>> suspectedLeaderTag() {
+        return _context -> ImmutableMap.of(
+                AtlasDbMetricNames.TAG_CURRENT_SUSPECTED_LEADER,
```

Comment:
Unrelated to PR: I am actually interested in seeing where this tag ends up being, if it's in metrics, it would be nice. I have not seen it used before (and was not able to easily get access to metrics production to check it's not filtered)

Reply:
You should be able to see these tags today for timelock services, this was just refactored to make reuse easier.

Reply:
example where they show up in tests:

https://github.com/palantir/atlasdb/blob/68587dca1addd7d501cb437b09f1c6154750f52a/timelock-agent/src/test/java/com/palantir/atlasdb/timelock/paxos/LeadershipComponentsTest.java#L157-L158